src/dcap_provider.cpp

```diff
@@ -69,12 +69,33 @@ static char PROCESSOR_CRL_NAME[] = "https%3a%2f%2fcertificates.trustedservices.i
 
 static std::string get_env_variable(std::string env_variable)
 {
+#ifdef __LINUX__
     const char* env_value = getenv(env_variable.c_str());
-
     if (env_value == NULL)
     {
         return std::string();
     }
+#else
+    char* env_value = (char*)malloc(MAX_ENV_VAR_LENGTH);

```

Comment:
Use unique_ptr<string> here so you don't have to worry about freeing it later on.

Reply:
thanks! I've implemented the change